implementations/rust/ockam/ockam/src/system/handler.rs

```diff
@@ -1,17 +1,47 @@
-use ockam_core::compat::boxed::Box;
-use ockam_core::{Message, Result, Routed};
+use dyn_clonable::*;
+use dyn_clone::DynClone;
+use ockam_core::compat::{boxed::Box, collections::BTreeMap, string::String};
+use ockam_core::{Address, Message, Result, Routed};
 
 /// Handle a single type of message for a worker system-address
 ///
 /// A handle may re-emit messages to the worker system, or to the
 /// Ockam runtime.  All state associated with a particular protocol
 /// must be contained in the type that implements this trait.
+#[clonable]
 #[ockam_core::async_trait]
-pub trait SystemHandler<C, M>
+pub trait SystemHandler<C, M>: Clone + DynClone
 where
     C: Send + 'static,
     M: Message,
 {
+    /// This function is called once during initialisation to setup
+    /// the internal route path
+    ///
+    /// A SystemHandler is able to send messages to both external
+    /// workers, and other internal handlers.  To allow workers to
+    /// create behaviour pipelines, we need to pre-define "routes" for
+    /// a SystemHandler (i.e. where is a message sent after it is done
+    /// processing it).
+    ///
+    /// In most cases this only requires a "default" route, but may
+    /// use different routing labels in more complicated setups (to
+    /// build processing graphs, instead of pipelines).
+    ///
+    /// It is highly recommended to use the
```

Comment:
Should this piece of documentation should go on the trait docs, rather than the function docs?